File: todo-core/todo.core.main/todo.core.services/src/main/java/todo/core/services/crud/AppCrudService.java
package todo.core.services.crud;

import todo.core.domain.BaseDomain;
import todo.core.domain.BaseDomainForm;

public interface AppCrudService<T extends BaseDomain, F extends BaseDomainForm> {

	T create(F domainForm) throws Exception;

	T update(F domainForm) throws Exception;

	void deleteDomainUsingPk(F domainForm) throws Exception;

	void deleteAll() throws Exception;

	T getDomainUsingPk(F domainForm) throws Exception;

	Iterable<T> getAll() throws Exception;

	default void batchCreate(F domainForm) throws Exception {

	}
}

File: todo-platform-ui/angular/todo-app/src/app/todo/todo-main/todo-core/services/crud/app-crud-service.ts
import { OnInit, Input } from '@angular/core';
import { Observable } from 'rxjs';
import { BaseDomain } from '../../domain/basedomain';
import { FormGroup } from '@angular/forms';

export class AppCrudService implements OnInit {

  @Input() public domainFormGroup: FormGroup;
  baseDomainItems: Observable<BaseDomain[]>;
  apiName: string;

  ngOnInit() {

  }

  create() {

  }

  update() {

  }

  deleteDomainUsingPk() {

  }

  deleteAll() {

  }

  getDomainUsingPk() {

  }


}